src/Rules/PassesConditions.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Enjin\Platform\Beam\Rules;
+
+use Closure;
+use Illuminate\Contracts\Validation\DataAwareRule;
+use Illuminate\Contracts\Validation\Rule;
+use Illuminate\Contracts\Validation\ValidationRule;
+
+class PassesConditions implements DataAwareRule, ValidationRule
+{
+    public static array $functions = [];
+
+    protected array $data = [];
+
+    /**
+     * Create new rule instance.
+     */
+    public function __construct(
+        protected bool $singleUse
+    ) {
+    }
+
+    /**
+     * Get the validation error message.
+     */
+    public function message()
+    {
+        return __('enjin-platform-beam::validation.passes_conditions');
+    }
+
+    public function validate(string $attribute, mixed $value, Closure $fail): void
+    {
+        $conditions = collect(static::$functions);
+
+        if (!$conditions->every(fn ($function) => $function($value, $this->singleUse, $this->data))) {
```

Comment:
I think we need to pass attributes as well, might have some use for some use cases